source/microbit/mprun.c

```diff
@@ -105,9 +118,11 @@ void mp_run(void) {
     readline_init0();
     microbit_init();
 
-    if (APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
+    if (hard_reboot && APPENDED_SCRIPT->header[0] == 'M' && APPENDED_SCRIPT->header[1] == 'P') {
```

Comment:
> But how do you get to the raw REPL?

Type ctrl-A at the REPL (either raw or friendly) (note: double ctrl-A if using picocom, since ctrl-A is picocom's escape code).

> Any attempt to leave the running script with ctrl-C will merely restart the script, so it impossible to enter the raw REPL.

Ctrl-C should drop you to the REPL, not restart the script.